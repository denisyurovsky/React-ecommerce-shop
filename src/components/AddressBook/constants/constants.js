export const LISTBOX_PADDING = 8;
export const DOUBLE_LISTBOX_PADDING = LISTBOX_PADDING * 2;
export const LIMIT_NUMBER_ITEMS = 8;
export const ITEM_SIZE_LG = 48;
export const ITEM_SIZE_MD = 36;
