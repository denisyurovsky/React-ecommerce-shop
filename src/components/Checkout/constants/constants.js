export const PANEL = {
  DELIVERY_ADDRESS: 'Delivery Address',
  PERSONAL_INFORMATION: 'Personal Information',
  PERSONAL_ADDRESS: 'Personal Address',
  PAYMENT_METHOD: 'Payment method',
  ORDER_CONFIRMATION: 'Order confirmation',
};
