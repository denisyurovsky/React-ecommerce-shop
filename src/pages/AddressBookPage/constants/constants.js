export const MODAL_TYPE = {
  EDIT: 'edit',
  ADD: 'add',
};

export const EMPTY_ADDRESS = {
  title: '',
  name: '',
  surname: '',
  country: { name: '', id: '' },
  city: '',
  street: '',
  building: '',
  flat: '',
  phone: '',
  zip: '',
};

export const LINKS = [
  { url: '/', text: 'Home' },
  { url: '/profile', text: 'Profile' },
  {
    url: `/profile/address-book`,
    text: 'Address book',
  },
];

export const RUSSIA = 'Russian Federation';
