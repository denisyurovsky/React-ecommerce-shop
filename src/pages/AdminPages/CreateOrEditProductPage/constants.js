export const PRICE_ERROR = 'Only numbers';
export const DESCRIPTION_ERROR = 'Only latin characters';
export const NOT_FOUND_PRODUCT_ERROR =
  'The item you are looking for was not found';
