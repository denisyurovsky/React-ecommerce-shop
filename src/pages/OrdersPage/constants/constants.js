export const breadcrumbsLinks = [
  { url: '/', text: 'Home' },
  { url: '/profile', text: 'Profile' },
  { url: '/profile/orders', text: 'My orders' },
];
