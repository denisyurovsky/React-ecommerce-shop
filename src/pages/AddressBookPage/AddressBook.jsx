import { Box, Button, Typography } from '@mui/material';
import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { toast } from 'react-toastify';

import AddressCard from '../../components/AddressBook/AddressCard/AddressCard';
import CreateOrEditAddressModal from '../../components/AddressBook/CreateOrEditAddressModal/CreateOrEditAddressModal';
import Breadcrumbs from '../../components/Breadcrumbs/Breadcrumbs';
import { Title } from '../../components/Title/Title';
import Spinner from '../../components/ui-kit/Spinner';
import { notificationError } from '../../helpers/constants/constants';
import usePrevious from '../../hooks/usePrevious';
import {
  getAddressesByIds,
  selectAddresses,
} from '../../store/addresses/addressesSlice';

import { EMPTY_ADDRESS, LINKS, MODAL_TYPE } from './constants/constants';

import styles from './AddressBook.module.scss';

const AddressBook = () => {
  const dispatch = useDispatch();
  const addresses = useSelector(selectAddresses);
  const user = useSelector((state) => state.user);
  const addressIds = user.user.addresses;

  const [isLoading, setIsLoading] = useState(true);
  const [address, setAddress] = useState(EMPTY_ADDRESS);
  const [isOpenModal, setIsOpenModal] = useState(false);
  const [modalType, setModalType] = useState(MODAL_TYPE.EDIT);

  useEffect(() => {
    dispatch(getAddressesByIds(addressIds));
  }, [addressIds, dispatch]);

  useEffect(() => {
    setIsLoading(addresses.isLoading || user.isLoading);
  }, [addresses.isLoading, user.isLoading]);

  const { errorOccurred: addressesErrorOccurred } = addresses;
  const { errorOccurred: userErrorOccurred } = user;
  const prevAmount = usePrevious({
    addressesErrorOccurred,
    userErrorOccurred,
  });

  useEffect(() => {
    const isErrorOccurred = addressesErrorOccurred || userErrorOccurred;

    if (prevAmount) {
      const isErrorOccurredPrev =
        prevAmount.addressesErrorOccurred || prevAmount.userErrorOccurred;

      if (isErrorOccurredPrev !== isErrorOccurred && isErrorOccurred) {
        toast.error(notificationError);
      }
    }
  }, [addressesErrorOccurred, userErrorOccurred, prevAmount]);

  const handleCloseModal = () => setIsOpenModal(false);
  const handleOpenModal = () => setIsOpenModal(true);
  const handleAddButton = () => {
    handleOpenModal();
    setModalType(MODAL_TYPE.ADD);
    setAddress(EMPTY_ADDRESS);
  };

  if (isLoading) {
    return <Spinner height="90vh" />;
  }

  return (
    <Box className={styles.container}>
      <Breadcrumbs links={LINKS} />
      <Title>Address book</Title>
      {addresses.data.length ? (
        addresses.data.map((address, id) => {
          return (
            <AddressCard
              key={id}
              address={address}
              handleOpenModal={handleOpenModal}
              setModalType={setModalType}
              setAddress={setAddress}
            />
          );
        })
      ) : (
        <Typography className={styles.title} variant="h4">
          There are no addresses
        </Typography>
      )}
      <Button
        onClick={handleAddButton}
        variant="contained"
        className={styles.button}
      >
        Add address
      </Button>
      <CreateOrEditAddressModal
        modalType={modalType}
        isOpenModal={isOpenModal}
        handleCloseModal={handleCloseModal}
        address={address}
        setAddress={setAddress}
      />
    </Box>
  );
};

export default AddressBook;
