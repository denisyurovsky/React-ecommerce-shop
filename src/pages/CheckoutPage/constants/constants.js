import { pathNames } from '../../../constants/pathNames';

export const LINKS = [
  { url: '/', text: 'Home' },
  {
    url: pathNames.CHECKOUT,
    text: 'Checkout',
  },
];

export const EMPTY_ORDER = {
  userId: null,
  id: null,
  products: [],
  status: null,
  addressId: null,
  deliveryAddress: {},
  deliveryPrice: null,
  totalPrice: null,
  totalDiscountPrice: null,
  totalQuantity: null,
  deliveredAt: null,
  createdAt: null,
  updatedAt: null,
};
