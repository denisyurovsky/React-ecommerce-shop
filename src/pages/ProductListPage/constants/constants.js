import { pathNames } from '../../../constants/pathNames';

export const sortTypes = {
  NEW_FIRST: 'new',
  OLD_FIRST: 'old',
  EXPENSIVE_FIRST: 'expensive',
  CHEAP_FIRST: 'cheap',
};

export const pageView = {
  MODULE_VIEW: 'module',
  LIST_VIEW: 'list',
};

export const sortObj = {
  [sortTypes.NEW_FIRST]: { field: 'createdAt', order: 'desc' },
  [sortTypes.OLD_FIRST]: { field: 'createdAt', order: 'asc' },
  [sortTypes.EXPENSIVE_FIRST]: { field: 'price', order: 'desc' },
  [sortTypes.CHEAP_FIRST]: { field: 'price', order: 'asc' },
};

export const BreadcrumbsLinks = [
  { url: '/', text: 'Home' },
  { url: pathNames.PRODUCTS, text: 'Products' },
];

export const NUMBER_ITEMS_ON_PAGE = 20;

export const MOBILE_WIDTH = 540;

export const defaultParams = {
  entity: 'products',
  filters: null,
  sort: sortObj[sortTypes.NEW_FIRST],
  currentPage: 1,
  itemsPerPage: NUMBER_ITEMS_ON_PAGE,
  q: '',
};
