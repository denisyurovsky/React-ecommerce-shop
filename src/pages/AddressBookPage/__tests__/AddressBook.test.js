import { waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import _ from 'lodash';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import React from 'react';

import { notificationError } from '../../../constants/constants';
import getAddressBookElements from '../../../test-utils/addressBook/getAddressBookElements';
import { addresses, newAddress } from '../../../test-utils/dto/addressesDto';
import { citiesRU, citiesBY } from '../../../test-utils/dto/citiesDto';
import { countries } from '../../../test-utils/dto/countriesDto';
import users from '../../../test-utils/dto/usersDto';
import render, { screen } from '../../../test-utils/renderWith';
import AddressBook from '../AddressBook';

const waitForAddressBook = () => screen.findAllByText(/address book/i);
const preloadedState = {
  user: users[0],
};

const server = setupServer(
  rest.get('/countries', (req, res, ctx) => res(ctx.json(countries))),
  rest.get('/cities/RU', (req, res, ctx) => res(ctx.json(citiesRU))),
  rest.get('/cities/BY', (req, res, ctx) => res(ctx.json(citiesBY))),
  rest.get('/addresses', (req, res, ctx) => {
    const addressIds = req.url.searchParams.getAll('id');
    const filteredAddresses = addressIds.map((id) => addresses[id - 1]);

    return res(ctx.json(filteredAddresses));
  }),
  rest.post('/addresses', (req, res, ctx) => {
    if (_.isEqual(newAddress, req.body)) {
      return res(ctx.json({ ...req.body, id: 4 }));
    }

    return res(ctx.status(400));
  }),
  rest.put('/addresses/1', (req, res, ctx) => {
    return res(ctx.json(req.body));
  }),
  rest.put('/addresses/3', (req, res, ctx) => {
    return res(ctx.status(400));
  }),
  rest.patch('/users/0', (req, res, ctx) => {
    return res(ctx.json(req.body));
  })
);

describe('AddressBook component', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  describe('Snapshot', () => {
    it('should render valid snapshot of AddressBook with data', async () => {
      const { asFragment } = render(<AddressBook />, preloadedState);

      await waitForAddressBook();
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Main group of tests', () => {
    beforeEach(() => {
      render(<AddressBook />, { user: users[0] });
    });

    it('should add new address correctly', async () => {
      await waitForAddressBook();
      userEvent.click(screen.getByRole('button', { name: /add address/i }));

      const {
        titleSelect,
        nameInput,
        surnameInput,
        phoneInput,
        autocompleteCountry,
        countryInput,
        autocompleteCity,
        cityInput,
        streetInput,
        buildingInput,
        flatInput,
        zipInput,
      } = getAddressBookElements(screen);

      userEvent.click(titleSelect);
      userEvent.click(await screen.findByRole('option', { name: 'Mr.' }));

      userEvent.type(nameInput, 'Jon');
      userEvent.type(surnameInput, 'Smith');
      userEvent.type(phoneInput, '79811112233');

      userEvent.type(countryInput, 'Russian F');
      await screen.findByRole('option', { name: /Russian Federation/i });
      userEvent.type(autocompleteCountry, '{arrowdown}');
      userEvent.type(autocompleteCountry, '{enter}');

      userEvent.type(cityInput, 'Saint P');
      await screen.findByRole('option', { name: /Saint Petersburg/i });
      userEvent.type(autocompleteCity, '{arrowdown}');
      userEvent.type(autocompleteCity, '{enter}');

      userEvent.type(streetInput, 'Nevskiy');
      userEvent.type(buildingInput, '8');
      userEvent.type(flatInput, '33');
      userEvent.type(zipInput, '168142');

      userEvent.click(screen.getByText('add'));

      expect(
        await screen.findByText(/Address was successfully added/i)
      ).toBeInTheDocument();

      await waitFor(
        async () => {
          expect(await screen.findByText(/Nevskiy/i)).toBeInTheDocument();
        },
        {
          timeout: 10000,
        }
      );
    });

    it('Should get an error after trying to add an address', async () => {
      await waitForAddressBook();
      userEvent.click(screen.getByRole('button', { name: /add address/i }));

      const {
        titleSelect,
        nameInput,
        surnameInput,
        phoneInput,
        autocompleteCountry,
        countryInput,
        autocompleteCity,
        cityInput,
        streetInput,
        buildingInput,
        flatInput,
        zipInput,
      } = getAddressBookElements(screen);

      userEvent.click(titleSelect);
      userEvent.click(await screen.findByRole('option', { name: 'Mr.' }));

      userEvent.type(nameInput, 'test');
      userEvent.type(surnameInput, 'test');
      userEvent.type(phoneInput, '11111111111');

      userEvent.type(countryInput, 'Russian F');
      await screen.findByRole('option', { name: /Russian Federation/i });
      userEvent.type(autocompleteCountry, '{arrowdown}');
      userEvent.type(autocompleteCountry, '{enter}');

      userEvent.type(cityInput, 'Saint P');
      await screen.findByRole('option', { name: /Saint Petersburg/i });
      userEvent.type(autocompleteCity, '{arrowdown}');
      userEvent.type(autocompleteCity, '{enter}');

      userEvent.type(streetInput, 'test');
      userEvent.type(buildingInput, 'test');
      userEvent.type(flatInput, 'test');
      userEvent.type(zipInput, '123456');

      userEvent.click(screen.getByText('add'));

      expect(await screen.findByText(notificationError)).toBeInTheDocument();
      expect(screen.queryByText(/test/i)).not.toBeInTheDocument();
    });

    it('should edit new address correctly', async () => {
      await waitForAddressBook();

      userEvent.click(screen.getAllByRole('button', { name: /edit/i })[0]);

      const {
        titleSelect,
        nameInput,
        surnameInput,
        phoneInput,
        autocompleteCountry,
        countryInput,
        autocompleteCity,
        cityInput,
        streetInput,
        buildingInput,
        flatInput,
        zipInput,
      } = getAddressBookElements(screen);

      userEvent.click(titleSelect);
      userEvent.click(await screen.findByRole('option', { name: 'Mrs.' }));

      userEvent.clear(nameInput);
      userEvent.type(nameInput, 'Alexandra');
      userEvent.clear(surnameInput);
      userEvent.type(surnameInput, 'Sidorova');
      userEvent.clear(phoneInput);
      userEvent.type(phoneInput, '37525002020');

      userEvent.clear(countryInput);
      userEvent.type(countryInput, 'Bela');
      await screen.findByRole('option', { name: /Belarus/i });
      userEvent.type(autocompleteCountry, '{arrowdown}');
      userEvent.type(autocompleteCountry, '{enter}');

      userEvent.type(cityInput, 'Min');
      await screen.findByRole('option', { name: /Minsk/i });
      userEvent.type(autocompleteCity, '{arrowdown}');
      userEvent.type(autocompleteCity, '{enter}');

      userEvent.clear(streetInput);
      userEvent.type(streetInput, 'Avangardnaya');
      userEvent.clear(buildingInput);
      userEvent.type(buildingInput, '12');
      userEvent.clear(flatInput);
      userEvent.type(flatInput, '21');
      userEvent.clear(zipInput);
      userEvent.type(zipInput, '220004');

      userEvent.click(screen.getByText('save'));

      expect(
        await screen.findByText(/Address was successfully edited/i)
      ).toBeInTheDocument();
      expect(await screen.findByText(/Avangardnaya/i)).toBeInTheDocument();
    });

    it('Should get an error after trying to edit an address', async () => {
      await waitForAddressBook();

      userEvent.click(screen.getAllByRole('button', { name: /edit/i })[1]);
      userEvent.click(screen.getByText('save'));

      expect(await screen.findByText(notificationError)).toBeInTheDocument();
    });
  });
});
