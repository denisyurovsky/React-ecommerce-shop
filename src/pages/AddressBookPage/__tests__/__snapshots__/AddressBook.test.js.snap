// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressBook component Snapshot should render valid snapshot of AddressBook with data 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="breadcrumbs"
      role="presentation"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/users"
            >
              Users
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/profile"
            >
              Profile
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="header"
    >
      <h2
        class="header"
        data-testid="title"
      >
        Address book
      </h2>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg container css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root container css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          data-testid="menuList"
          id="composition-menu"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <a
              class="link grey"
              href="/profile"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Profile
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/wishlist"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FavoriteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Wishlist
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/orders"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="Inventory2Icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 2H4c-1 0-2 .9-2 2v3.01c0 .72.43 1.34 1 1.69V20c0 1.1 1.1 2 2 2h14c.9 0 2-.9 2-2V8.7c.57-.35 1-.97 1-1.69V4c0-1.1-1-2-2-2zm-5 12H9v-2h6v2zm5-7H4V4h16v3z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Orders
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/address-book"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ImportContactsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.5 4.5c-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .65.73.45.75.45C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.41.21.75-.19.75-.45V6c-1.49-1.12-3.63-1.5-5.5-1.5zm3.5 14c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Address book
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root profile css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="container MuiBox-root css-0"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium closeIcon css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Mr. Alexander Sidorov
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                +7 (981) 3002020
              </p>
              <br />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Russian Federation, Saint Petersburg
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Pulkovskaya, 8, 4
              </p>
              <br />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                168142
              </p>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing footer css-1t6e9jv-MuiCardActions-root"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Edit
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium closeIcon css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Mrs. Olga Petrova
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                +7 (981) 1233232
              </p>
              <br />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Russian Federation, Saint Petersburg
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Pobedy, 8, 4
              </p>
              <br />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                168142
              </p>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing footer css-1t6e9jv-MuiCardActions-root"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Edit
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root button css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Add address
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
